.github/workflows/pr.yml

```diff
@@ -44,36 +44,59 @@ jobs:
     uses: ./.github/workflows/dispatch-build-and-test.yml
     strategy:
       fail-fast: false
-      matrix: 
-        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }} 
+      matrix:
+        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }}
         compiler: ${{ fromJSON(needs.compute-matrix.outputs.COMPILERS) }}
     with:
       per_cuda_compiler_matrix: ${{ toJSON(fromJSON(needs.compute-matrix.outputs.PER_CUDA_COMPILER_MATRIX)[ format('{0}-{1}', matrix.cuda_version, matrix.compiler) ]) }}
       build_script: "./ci/build_cub.sh"
       test_script: "./ci/test_cub.sh"
-  
+
   libcudacxx:
     name: libcudacxx CUDA${{ matrix.cuda_version }} ${{ matrix.compiler }}
     needs: compute-matrix
     uses: ./.github/workflows/dispatch-build-and-test.yml
     strategy:
       fail-fast: false
-      matrix: 
-        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }} 
+      matrix:
+        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }}
         compiler: ${{ fromJSON(needs.compute-matrix.outputs.COMPILERS) }}
     with:
       per_cuda_compiler_matrix: ${{ toJSON(fromJSON(needs.compute-matrix.outputs.PER_CUDA_COMPILER_MATRIX)[ format('{0}-{1}', matrix.cuda_version, matrix.compiler) ]) }}
       build_script: "./ci/build_libcudacxx.sh"
-      test_script: "./ci/test_libcudacxx.sh" 
+      test_script: "./ci/test_libcudacxx.sh"
+
+  nvrtc:
+    name: libcudacxx nvrtc
+    needs: compute-matrix
+    uses: ./.github/workflows/dispatch-build-and-test.yml
+    strategy:
+      fail-fast: false
+      matrix:
+        cuda_version: ${{ fromJSON(needs.compute-matrix.outputs.CUDA_VERSIONS) }}
+        compiler: ${{ fromJSON(needs.compute-matrix.outputs.COMPILERS) }}
+        exclude:
+          - cuda_version: "11.1"
+          - cuda_version: "12.1"
+          - compiler: clang
+          - compiler: gcc
+        include:
+          - cuda_version: "12.1"
+          - compiler: gcc-12
```

Comment:
This is effectively the same as just hardcoding the environments to use. Instead, we want to query from the matrix file what the newest CUDA version is and use that. I think it will be easier to not use the `build-and-test.yaml` workflow.

Reply:
https://github.com/NVIDIA/cccl/issues/204

I'm going to do this first which will make this PR easier.